dev-java/javacup: add 11b_p20160615

vaukai opened the pull request:
Closes: https://bugs.gentoo.org/588824

fordfrog commented:
i can't compile it.

with openjdk-17:
```
2023-09-16 13:57:13 (834 KB/s) - ‘/usr/src/gentoo.git/distfiles/java-cup-src-11b-20160615.tar.gz.__download__’ saved [313847/313847]

 * java-cup-src-11b-20160615.tar.gz BLAKE2B SHA512 size ;-) ...                                                                                                                                                                                                                                                                                                     [ ok ]
 * Using: openjdk-17
>>> Unpacking source...
>>> Unpacking java-cup-src-11b-20160615.tar.gz to /var/tmp/portage/dev-java/javacup-11b_p20160615/work
>>> Source unpacked in /var/tmp/portage/dev-java/javacup-11b_p20160615/work
>>> Preparing source in /var/tmp/portage/dev-java/javacup-11b_p20160615/work ...
 * Applying javacup-11b_p20160615-build-xml-svn.patch ...                                                                                                                                                                                                                                                                                                           [ ok ]
removed 'bin/JFlex.jar'
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/dev-java/javacup-11b_p20160615/work ...
Rewriting attributes
Rewriting ./build.xml
>>> Source configured.
>>> Compiling source in /var/tmp/portage/dev-java/javacup-11b_p20160615/work ...
 * Bootstrapping with bundled javacup ...
 * Disabling all optional ANT_TASKS
Warning: Unable to determine tools.jar location.
  If build fails because sun.* classes could not be found,
  Make sure you are using a JDK, not JRE as your user/system VM.
  and that you have java-config version 2.0.30 or above installed.
Buildfile: /var/tmp/portage/dev-java/javacup-11b_p20160615/work/build.xml

init:
     [exec] Result: 128
     [exec] Result: 128

cup:
      [cup] This is CUP v0.11a beta 20060608
      [cup] Authors : Scott E. Hudson, Frank Flannery, Andrea Flexeder, Michael Petter and C. Scott Ananian
      [cup] Bugreports to petter@cs.tum.edu
      [cup] Destination directory didn't exist; creating new one: /var/tmp/portage/dev-java/javacup-11b_p20160615/work/java/java_cup
      [cup] ------- CUP v0.11a beta 20060608 Parser Generation Summary -------
      [cup]   0 errors and 0 warnings
      [cup]   36 terminals, 60 non-terminals, and 117 productions declared, 
      [cup]   producing 185 unique parse states.
      [cup]   0 terminals declared but not used.
      [cup]   0 non-terminals declared but not used.
      [cup]   0 productions never reduced.
      [cup]   0 conflicts detected (0 expected).
      [cup]   Code written to "parser.java", and "sym.java".
      [cup] ---------------------------------------------------- (v0.11a beta 20060608)

jflex:

BUILD FAILED
/var/tmp/portage/dev-java/javacup-11b_p20160615/work/build.xml:76: java.lang.NoSuchMethodError: 'java_cup.runtime.Symbol java_cup.runtime.SymbolFactory.newSymbol(java.lang.String, int, java_cup.runtime.Symbol, java.lang.Object)'
	at jflex.LexParse$CUP$LexParse$actions.CUP$LexParse$do_action_part00000000(LexParse.java:1415)
	at jflex.LexParse$CUP$LexParse$actions.CUP$LexParse$do_action(LexParse.java:3399)
	at jflex.LexParse.do_action(LexParse.java:940)
	at java_cup.runtime.lr_parser.parse(lr_parser.java:584)
	at jflex.Main.generate(Main.java:67)
	at jflex.anttask.JFlexTask.execute(JFlexTask.java:66)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
	at jdk.internal.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at org.apache.tools.ant.Target.execute(Target.java:449)
	at org.apache.tools.ant.Target.performTasks(Target.java:470)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1401)
	at org.apache.tools.ant.Project.executeTarget(Project.java:1374)
	at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
	at org.apache.tools.ant.Project.executeTargets(Project.java:1264)
	at org.apache.tools.ant.Main.runBuild(Main.java:827)
	at org.apache.tools.ant.Main.startAnt(Main.java:223)
	at org.apache.tools.ant.launch.Launcher.run(Launcher.java:284)
	at org.apache.tools.ant.launch.Launcher.main(Launcher.java:101)
```
with openjdk-8:
```
>>> Emerging (1 of 1) dev-java/javacup-11b_p20160615::gentoo
 * java-cup-src-11b-20160615.tar.gz BLAKE2B SHA512 size ;-) ...                                                                                                                                                                                                                                                                                                     [ ok ]
 * Using: openjdk-8
>>> Unpacking source...
>>> Unpacking java-cup-src-11b-20160615.tar.gz to /var/tmp/portage/dev-java/javacup-11b_p20160615/work
>>> Source unpacked in /var/tmp/portage/dev-java/javacup-11b_p20160615/work
>>> Preparing source in /var/tmp/portage/dev-java/javacup-11b_p20160615/work ...
 * Applying javacup-11b_p20160615-build-xml-svn.patch ...                                                                                                                                                                                                                                                                                                           [ ok ]
removed 'bin/JFlex.jar'
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/dev-java/javacup-11b_p20160615/work ...
Rewriting attributes
Rewriting ./build.xml
>>> Source configured.
>>> Compiling source in /var/tmp/portage/dev-java/javacup-11b_p20160615/work ...
 * Bootstrapping with bundled javacup ...
 * Disabling all optional ANT_TASKS
Buildfile: /var/tmp/portage/dev-java/javacup-11b_p20160615/work/build.xml

init:
     [exec] Result: 128

BUILD FAILED
/var/tmp/portage/dev-java/javacup-11b_p20160615/work/build.xml:30: java.lang.NoSuchMethodError: java.nio.CharBuffer.flip()Ljava/nio/CharBuffer;
	at org.apache.tools.ant.util.ReaderInputStream.<init>(ReaderInputStream.java:81)
	at org.apache.tools.ant.util.ReaderInputStream.<init>(ReaderInputStream.java:66)
	at org.apache.tools.ant.util.ReaderInputStream.<init>(ReaderInputStream.java:118)
	at org.apache.tools.ant.util.ReaderInputStream.<init>(ReaderInputStream.java:105)
	at org.apache.tools.ant.taskdefs.Redirector.createStreams(Redirector.java:594)
	at org.apache.tools.ant.taskdefs.Redirector.createHandler(Redirector.java:765)
	at org.apache.tools.ant.taskdefs.ExecTask.createHandler(ExecTask.java:690)
	at org.apache.tools.ant.taskdefs.ExecTask.prepareExec(ExecTask.java:601)
	at org.apache.tools.ant.taskdefs.ExecTask.execute(ExecTask.java:497)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
	at org.apache.tools.ant.Task.perform(Task.java:350)
	at org.apache.tools.ant.Target.execute(Target.java:449)
	at org.apache.tools.ant.Target.performTasks(Target.java:470)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1401)
	at org.apache.tools.ant.Project.executeTarget(Project.java:1374)
	at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
	at org.apache.tools.ant.Project.executeTargets(Project.java:1264)
	at org.apache.tools.ant.Main.runBuild(Main.java:827)
	at org.apache.tools.ant.Main.startAnt(Main.java:223)
	at org.apache.tools.ant.launch.Launcher.run(Launcher.java:284)
	at org.apache.tools.ant.launch.Launcher.main(Launcher.java:101)
```

vaukai commented:
> i can't compile it.

Sorry, cannot reproduce those errors. Have to investigate.

vaukai commented:
switched to building with `java-pkg-simple.eclass` using `>=dev-java/ant-1.10.14` instead of old `ant-core`

fordfrog commented:
i get an error:
```
>>> Emerging (1 of 1) dev-java/javacup-11b_p20160615::gentoo
 * java-cup-src-11b-20160615.tar.gz BLAKE2B SHA512 size ;-) ...                                                                                                                                                                                                                                                                                                     [ ok ]
 * Using: openjdk-21
>>> Unpacking source...
>>> Unpacking java-cup-src-11b-20160615.tar.gz to /var/tmp/portage/dev-java/javacup-11b_p20160615/work
>>> Source unpacked in /var/tmp/portage/dev-java/javacup-11b_p20160615/work
>>> Preparing source in /var/tmp/portage/dev-java/javacup-11b_p20160615/work ...
Searching for bundled jars:
/var/tmp/portage/dev-java/javacup-11b_p20160615/work/bin/JFlex.jar
/var/tmp/portage/dev-java/javacup-11b_p20160615/work/bin/java-cup-11.jar
Searching for bundled classes (no output if none found):
Search done.
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/dev-java/javacup-11b_p20160615/work ...
>>> Source configured.
>>> Compiling source in /var/tmp/portage/dev-java/javacup-11b_p20160615/work ...
 * Get path of bundled JAVACUP
 * Path to JAVACUP is bin/java-cup-11.jar
 * Get path to jflex
 * Path to jflex is /usr/share/jflex/lib/jflex.jar
 * Generate parser with: bin/java-cup-11.jar
------- CUP v0.11a beta 20060608 Parser Generation Summary -------
  0 errors and 0 warnings
  36 terminals, 60 non-terminals, and 117 productions declared, 
  producing 185 unique parse states.
  0 terminals declared but not used.
  0 non-terminals declared but not used.
  0 productions never reduced.
  0 conflicts detected (0 expected).
  Code written to "parser.java", and "sym.java".
---------------------------------------------------- (v0.11a beta 20060608)
 * Generate lexer
Reading "flex/Lexer.jflex"
Exception in thread "main" java.lang.NoSuchMethodError: 'java_cup.runtime.Symbol java_cup.runtime.SymbolFactory.newSymbol(java.lang.String, int, java_cup.runtime.Symbol, java.lang.Object)'
	at jflex.LexParse$CUP$LexParse$actions.CUP$LexParse$do_action_part00000000(LexParse.java:1415)
	at jflex.LexParse$CUP$LexParse$actions.CUP$LexParse$do_action(LexParse.java:3399)
	at jflex.LexParse.do_action(LexParse.java:940)
	at java_cup.runtime.lr_parser.parse(lr_parser.java:584)
	at jflex.Main.generate(Main.java:67)
	at jflex.Main.generate(Main.java:394)
	at jflex.Main.main(Main.java:411)
```

vaukai commented:
> i get an error:

I don't understand why that happens to you. But the error message you are getting looks similar to what's reported in https://bugs.gentoo.org/776655 for the old version built with `java-ant-2.eclass`

I cannot reproduce that error, for me it works. What I am getting instead of your error message is:
```
---------------------------------------------------- (v0.11a beta 20060608)
 [32m*[0m Generate lexer
Reading "flex/Lexer.jflex"

Warning in file "flex/Lexer.jflex" (line 102): 
".|\n" does not match all characters, because "." excludes all Unicode newline chars - use "[^]" instead
  .|\n            { sb.append(yytext()); }
^

Warning in file "flex/Lexer.jflex" (line 106): 
".|\n" does not match all characters, because "." excludes all Unicode newline chars - use "[^]" instead
.|\n          { emit_warning("Unrecognized character '" +yytext()+"' -- ignored"); }
^
Constructing NFA : 241 states in NFA
Converting NFA to DFA : 
...............................................................................................................................................
147 states before minimization, 132 states in minimized DFA
Writing code to "java/javacup/Lexer.java"
```
The full log of my successsful build is here:
```
[32m * [39;49;00mPackage:    dev-java/javacup-11b_p20160615:0
[32m * [39;49;00mRepository: testing
[32m * [39;49;00mMaintainer: java@gentoo.org
[32m * [39;49;00mUSE:        abi_x86_64 amd64 elibc_glibc kernel_linux
[32m * [39;49;00mFEATURES:   keeptemp network-sandbox preserve-libs sandbox userpriv usersandbox
debug:   eclass exists: /var/db/repos/gentoo/eclass/java-pkg-2.eclass
debug: inherit: java-pkg-2 -> /var/db/repos/gentoo/eclass/java-pkg-2.eclass
debug: *** Multiple Inheritance (Level: 2)
debug:   eclass exists: /var/db/repos/gentoo/eclass/java-utils-2.eclass
debug: inherit: java-utils-2 -> /var/db/repos/gentoo/eclass/java-utils-2.eclass
debug: EXPORT_FUNCTIONS: pkg_setup -> java-pkg-2_pkg_setup
debug: EXPORT_FUNCTIONS: src_prepare -> java-pkg-2_src_prepare
debug: EXPORT_FUNCTIONS: src_compile -> java-pkg-2_src_compile
debug: EXPORT_FUNCTIONS: pkg_preinst -> java-pkg-2_pkg_preinst
debug:   eclass exists: /var/db/repos/gentoo/eclass/java-pkg-simple.eclass
debug: inherit: java-pkg-simple -> /var/db/repos/gentoo/eclass/java-pkg-simple.eclass
debug: *** Multiple Inheritance (Level: 2)
debug:   eclass exists: /var/db/repos/gentoo/eclass/java-utils-2.eclass
debug: inherit: java-utils-2 -> /var/db/repos/gentoo/eclass/java-utils-2.eclass
debug: EXPORT_FUNCTIONS: src_compile -> java-pkg-simple_src_compile
debug: EXPORT_FUNCTIONS: src_install -> java-pkg-simple_src_install
debug: EXPORT_FUNCTIONS: src_test -> java-pkg-simple_src_test
debug: java-pkg_init: entering function, parameters: 
debug: java-pkg_switch-vm: entering function, parameters: 
debug: java-pkg_needs-vm: entering function, parameters: 
debug: depend-java-query:  NV_DEPEND:	>=dev-java/ant-1.10.14:0
	dev-java/jflex:0
	>=virtual/jdk-1.8:* >=dev-java/java-config-2.2.0-r3 source? ( app-arch/zip )
debug: Using: openjdk-bin-21
debug: java-pkg_setup-vm: entering function, parameters: 
debug: java-pkg_get-vm-vendor: entering function, parameters: 
debug: java-pkg_javac-args: entering function, parameters: 
debug: want source: 1.8
debug: want target: 1.8
debug: java-pkg_append_: entering function, parameters: LD_LIBRARY_PATH /opt/openjdk-bin-21.0.1_p12/lib/:/opt/openjdk-bin-21.0.1_p12/lib/server/
 [32m*[0m Using: openjdk-bin-21
>>> Unpacking source...
>>> Unpacking java-cup-src-11b-20160615.tar.gz to /var/tmp/portage/dev-java/javacup-11b_p20160615/work
>>> Source unpacked in /var/tmp/portage/dev-java/javacup-11b_p20160615/work
>>> Preparing source in /var/tmp/portage/dev-java/javacup-11b_p20160615/work ...
Searching for bundled jars:
/var/tmp/portage/dev-java/javacup-11b_p20160615/work/bin/JFlex.jar
/var/tmp/portage/dev-java/javacup-11b_p20160615/work/bin/java-cup-11.jar
Searching for bundled classes (no output if none found):
Search done.
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/dev-java/javacup-11b_p20160615/work ...
>>> Source configured.
>>> Compiling source in /var/tmp/portage/dev-java/javacup-11b_p20160615/work ...
 [32m*[0m Get path of bundled JAVACUP
 [32m*[0m Path to JAVACUP is bin/java-cup-11.jar
 [32m*[0m Get path to jflex
debug: java-pkg_getjars: entering function, parameters: --build-only jflex
debug: jflex:/usr/share/jflex/lib/jflex.jar
debug: java-pkg_ensure-dep: entering function, parameters: build jflex
debug: Matching against: /(jflex(-[^:]+)?:0|(-[^:]+)?:jflex)\b
debug: java-pkg_record-jar_: entering function, parameters: --build-only jflex
 [32m*[0m Path to jflex is /usr/share/jflex/lib/jflex.jar
 [32m*[0m Generate parser with: bin/java-cup-11.jar
------- CUP v0.11a beta 20060608 Parser Generation Summary -------
  0 errors and 0 warnings
  36 terminals, 60 non-terminals, and 117 productions declared, 
  producing 185 unique parse states.
  0 terminals declared but not used.
  0 non-terminals declared but not used.
  0 productions never reduced.
  0 conflicts detected (0 expected).
  Code written to "parser.java", and "sym.java".
---------------------------------------------------- (v0.11a beta 20060608)
 [32m*[0m Generate lexer
Reading "flex/Lexer.jflex"

Warning in file "flex/Lexer.jflex" (line 102): 
".|\n" does not match all characters, because "." excludes all Unicode newline chars - use "[^]" instead
  .|\n            { sb.append(yytext()); }
^

Warning in file "flex/Lexer.jflex" (line 106): 
".|\n" does not match all characters, because "." excludes all Unicode newline chars - use "[^]" instead
.|\n          { emit_warning("Unrecognized character '" +yytext()+"' -- ignored"); }
^
Constructing NFA : 241 states in NFA
Converting NFA to DFA : 
...............................................................................................................................................
147 states before minimization, 132 states in minimized DFA
Writing code to "java/javacup/Lexer.java"
 [32m*[0m Build intermediate jar
debug: java-pkg_gen-cp: entering function, parameters: JAVA_GENTOO_CLASSPATH
debug: java-pkg-simple_getclasspath: entering function, parameters: 
debug: java-pkg_getjars: entering function, parameters: --with-dependencies ant
debug: ant:/usr/share/ant/lib/ant.jar:/usr/share/ant/lib/ant-launcher.jar:/usr/share/ant/lib/ant-antlr.jar:/usr/share/ant/lib/ant-imageio.jar:/usr/share/ant/lib/ant-jmf.jar:/usr/share/ant/lib/ant-swing.jar
debug: java-pkg_ensure-dep: entering function, parameters: ant
debug: Matching against: /(ant(-[^:]+)?:0|(-[^:]+)?:ant)\b
debug: java-pkg_record-jar_: entering function, parameters: ant
debug: CLASSPATH=/usr/share/ant/lib/ant.jar:/usr/share/ant/lib/ant-launcher.jar:/usr/share/ant/lib/ant-antlr.jar:/usr/share/ant/lib/ant-imageio.jar:/usr/share/ant/lib/ant-jmf.jar:/usr/share/ant/lib/ant-swing.jar
debug: java-pkg-simple_prepend_resources: entering function, parameters: target/classes
debug: ejavac: entering function, parameters: -d target/classes -encoding UTF-8 -classpath /usr/share/ant/lib/ant.jar:/usr/share/ant/lib/ant-launcher.jar:/usr/share/ant/lib/ant-antlr.jar:/usr/share/ant/lib/ant-imageio.jar:/usr/share/ant/lib/ant-jmf.jar:/usr/share/ant/lib/ant-swing.jar @sources.lst
debug: java-pkg_get-javac: entering function, parameters: 
debug: java-pkg_init-compiler_: entering function, parameters: 
debug: Read "" from /etc/java-config-2/build/compilers.conf
debug: java-pkg_javac-args: entering function, parameters: 
debug: want source: 1.8
debug: want target: 1.8
 [32m*[0m Verbose logging for "ejavac" function
 [32m*[0m Compiler executable: javac
 [32m*[0m Extra arguments: -source 1.8 -target 1.8
 [32m*[0m Complete command:
 [32m*[0m javac -source 1.8 -target 1.8 -d target/classes -encoding UTF-8 -classpath /usr/share/ant/lib/ant.jar:/usr/share/ant/lib/ant-launcher.jar:/usr/share/ant/lib/ant-antlr.jar:/usr/share/ant/lib/ant-imageio.jar:/usr/share/ant/lib/ant-jmf.jar:/usr/share/ant/lib/ant-swing.jar @sources.lst
javac -source 1.8 -target 1.8 -d target/classes -encoding UTF-8 -classpath /usr/share/ant/lib/ant.jar:/usr/share/ant/lib/ant-launcher.jar:/usr/share/ant/lib/ant-antlr.jar:/usr/share/ant/lib/ant-imageio.jar:/usr/share/ant/lib/ant-jmf.jar:/usr/share/ant/lib/ant-swing.jar @sources.lst
warning: [options] bootstrap class path not set in conjunction with -source 8
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
src/java_cup/emit.java:410: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
		     pre("do_action_part")+ String.format("%08d",new Integer(instancecounter)) +"(");
		                                                 ^
src/java_cup/emit.java:577: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
	  out.println("              return " + pre("do_action_part")+ String.format("%08d",new Integer(0))+"(");
	                                                                                    ^
src/java_cup/emit.java:602: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
	  out.println("              return " + pre("do_action_part")+ String.format("%08d",new Integer(instancecounter))+"(");
	                                                                                    ^
src/java_cup/emit.java:1097: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
		     pre("do_action_part")+ String.format("%08d",new Integer(instancecounter)) +"(");
		                                                 ^
src/java_cup/emit.java:1219: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
	  out.println("              return " + pre("do_action_part")+ String.format("%08d",new Integer(0))+"(");
	                                                                                    ^
src/java_cup/emit.java:1244: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
	  out.println("              return " + pre("do_action_part")+ String.format("%08d",new Integer(instancecounter))+"(");
	                                                                                    ^
src/java_cup/production.java:157: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
      _all.put(new Integer(_index),this);
               ^
src/java_cup/production.java:230: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    return (production) _all.get(new Integer(indx));
                                 ^
src/java_cup/lalr_item_set.java:347: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
	  hashcode_cache = new Integer(result);
	                   ^
src/java_cup/non_terminal.java:43: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
      _all_by_index.put(new Integer(_index), this);
                        ^
src/java_cup/non_terminal.java:93: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
      Integer the_indx = new Integer(indx);
                         ^
src/java_cup/terminal.java:48: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
      _all_by_index.put(new Integer(_index), this);
                        ^
src/java_cup/terminal.java:117: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
      Integer the_indx = new Integer(indx);
                         ^
src/java_cup/runtime/virtual_parse_stack.java:93: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
      vstack.push(new Integer(stack_sym.parse_state));
                  ^
src/java_cup/runtime/virtual_parse_stack.java:140: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
      vstack.push(new Integer(state_num));
                  ^
src/java_cup/runtime/lr_parser.java:480: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
		  ret.add(new Integer(row[i]));
		          ^
Note: java/parser.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
20 warnings
updated manifest
 [32m*[0m Remove classes and generated code
removed 'java/javacup/Lexer.java'
removed directory 'java/javacup'
removed 'java/parser.java'
removed 'java/sym.java'
removed 'target/classes/java_cup/ErrorManager.class'
removed 'target/classes/java_cup/action_production.class'
removed 'target/classes/java_cup/production_part.class'
removed 'target/classes/java_cup/parse_action.class'
removed 'target/classes/java_cup/parser.class'
removed 'target/classes/java_cup/parse_reduce_row.class'
removed 'target/classes/java_cup/production.class'
removed 'target/classes/java_cup/sym.class'
removed 'target/classes/java_cup/reduce_action.class'
removed 'target/classes/java_cup/symbol_part.class'
removed 'target/classes/java_cup/lalr_transition.class'
removed 'target/classes/java_cup/parse_reduce_table.class'
removed 'target/classes/java_cup/lalr_item_set.class'
removed 'target/classes/java_cup/symbol.class'
removed 'target/classes/java_cup/Main.class'
removed 'target/classes/java_cup/Lexer.class'
removed 'target/classes/java_cup/non_terminal.class'
removed 'target/classes/java_cup/emit.class'
removed 'target/classes/java_cup/lalr_item.class'
removed 'target/classes/java_cup/terminal.class'
removed 'target/classes/java_cup/runtime/Symbol.class'
removed 'target/classes/java_cup/runtime/XMLElement$Terminal.class'
removed 'target/classes/java_cup/runtime/SyntaxTreeXPath.class'
removed 'target/classes/java_cup/runtime/XMLElement$NonTerminal.class'
removed 'target/classes/java_cup/runtime/SyntaxTreeTransform.class'
removed 'target/classes/java_cup/runtime/SymbolFactory.class'
removed 'target/classes/java_cup/runtime/virtual_parse_stack.class'
removed 'target/classes/java_cup/runtime/XMLElement.class'
removed 'target/classes/java_cup/runtime/SyntaxTreeDFS$Visitor.class'
removed 'target/classes/java_cup/runtime/lr_parser.class'
removed 'target/classes/java_cup/runtime/SyntaxTreeTransform$1.class'
removed 'target/classes/java_cup/runtime/XMLElement$Error.class'
removed 'target/classes/java_cup/runtime/ComplexSymbolFactory$ComplexSymbol.class'
removed 'target/classes/java_cup/runtime/SyntaxTreeDFS$ElementHandler.class'
removed 'target/classes/java_cup/runtime/SyntaxTreeTransform$ChainRemover.class'
removed 'target/classes/java_cup/runtime/ComplexSymbolFactory.class'
removed 'target/classes/java_cup/runtime/SyntaxTreeDFS.class'
removed 'target/classes/java_cup/runtime/SyntaxTreeDFS$AbstractVisitor.class'
removed 'target/classes/java_cup/runtime/DefaultSymbolFactory.class'
removed 'target/classes/java_cup/runtime/Scanner.class'
removed 'target/classes/java_cup/runtime/ComplexSymbolFactory$Location.class'
removed 'target/classes/java_cup/runtime/ScannerBuffer.class'
removed directory 'target/classes/java_cup/runtime'
removed 'target/classes/java_cup/terminal_set.class'
removed 'target/classes/java_cup/internal_error.class'
removed 'target/classes/java_cup/parse_action_row.class'
removed 'target/classes/java_cup/version.class'
removed 'target/classes/java_cup/CUP$parser$actions.class'
removed 'target/classes/java_cup/shift_action.class'
removed 'target/classes/java_cup/action_part.class'
removed 'target/classes/java_cup/symbol_set.class'
removed 'target/classes/java_cup/nonassoc_action.class'
removed 'target/classes/java_cup/parse_action_table.class'
removed 'target/classes/java_cup/lr_item_core.class'
removed 'target/classes/java_cup/assoc.class'
removed 'target/classes/java_cup/lalr_state.class'
removed 'target/classes/java_cup/anttask/CUPTask.class'
removed directory 'target/classes/java_cup/anttask'
removed directory 'target/classes/java_cup'
removed directory 'target/classes/'
 [32m*[0m Regenerate parser with intermediate jar file
------- CUP v0.11b beta 20140226 Parser Generation Summary -------
  0 errors and 0 warnings
  36 terminals, 60 non-terminals, and 117 productions declared, 
  producing 185 unique parse states.
  0 terminals declared but not used.
  0 non-terminals declared but not used.
  0 productions never reduced.
  0 conflicts detected (0 expected).
  Code written to "parser.java", and "sym.java".
---------------------------------------------------- (CUP v0.11b beta 20140226)
 [32m*[0m Regenerate lexer with intermediate jar file
Reading "flex/Lexer.jflex"

Warning in file "flex/Lexer.jflex" (line 102): 
".|\n" does not match all characters, because "." excludes all Unicode newline chars - use "[^]" instead
  .|\n            { sb.append(yytext()); }
^

Warning in file "flex/Lexer.jflex" (line 106): 
".|\n" does not match all characters, because "." excludes all Unicode newline chars - use "[^]" instead
.|\n          { emit_warning("Unrecognized character '" +yytext()+"' -- ignored"); }
^
Constructing NFA : 241 states in NFA
Converting NFA to DFA : 
...............................................................................................................................................
147 states before minimization, 132 states in minimized DFA
Writing code to "java/javacup/Lexer.java"
 [32m*[0m Build final jar with regenerated parser and lexer
debug: java-pkg_gen-cp: entering function, parameters: JAVA_GENTOO_CLASSPATH
debug: java-pkg-simple_getclasspath: entering function, parameters: 
debug: java-pkg_getjars: entering function, parameters: --with-dependencies ant
debug: ant:/usr/share/ant/lib/ant.jar:/usr/share/ant/lib/ant-launcher.jar:/usr/share/ant/lib/ant-antlr.jar:/usr/share/ant/lib/ant-imageio.jar:/usr/share/ant/lib/ant-jmf.jar:/usr/share/ant/lib/ant-swing.jar
debug: java-pkg_ensure-dep: entering function, parameters: ant
debug: Matching against: /(ant(-[^:]+)?:0|(-[^:]+)?:ant)\b
debug: java-pkg_record-jar_: entering function, parameters: ant
debug: CLASSPATH=/usr/share/ant/lib/ant.jar:/usr/share/ant/lib/ant-launcher.jar:/usr/share/ant/lib/ant-antlr.jar:/usr/share/ant/lib/ant-imageio.jar:/usr/share/ant/lib/ant-jmf.jar:/usr/share/ant/lib/ant-swing.jar
debug: java-pkg-simple_prepend_resources: entering function, parameters: target/classes
debug: ejavac: entering function, parameters: -d target/classes -encoding UTF-8 -classpath /usr/share/ant/lib/ant.jar:/usr/share/ant/lib/ant-launcher.jar:/usr/share/ant/lib/ant-antlr.jar:/usr/share/ant/lib/ant-imageio.jar:/usr/share/ant/lib/ant-jmf.jar:/usr/share/ant/lib/ant-swing.jar @sources.lst
debug: java-pkg_get-javac: entering function, parameters: 
debug: java-pkg_init-compiler_: entering function, parameters: 
debug: Read "" from /etc/java-config-2/build/compilers.conf
debug: java-pkg_javac-args: entering function, parameters: 
debug: want source: 1.8
debug: want target: 1.8
 [32m*[0m Verbose logging for "ejavac" function
 [32m*[0m Compiler executable: javac
 [32m*[0m Extra arguments: -source 1.8 -target 1.8
 [32m*[0m Complete command:
 [32m*[0m javac -source 1.8 -target 1.8 -d target/classes -encoding UTF-8 -classpath /usr/share/ant/lib/ant.jar:/usr/share/ant/lib/ant-launcher.jar:/usr/share/ant/lib/ant-antlr.jar:/usr/share/ant/lib/ant-imageio.jar:/usr/share/ant/lib/ant-jmf.jar:/usr/share/ant/lib/ant-swing.jar @sources.lst
javac -source 1.8 -target 1.8 -d target/classes -encoding UTF-8 -classpath /usr/share/ant/lib/ant.jar:/usr/share/ant/lib/ant-launcher.jar:/usr/share/ant/lib/ant-antlr.jar:/usr/share/ant/lib/ant-imageio.jar:/usr/share/ant/lib/ant-jmf.jar:/usr/share/ant/lib/ant-swing.jar @sources.lst
warning: [options] bootstrap class path not set in conjunction with -source 8
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
src/java_cup/emit.java:410: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
		     pre("do_action_part")+ String.format("%08d",new Integer(instancecounter)) +"(");
		                                                 ^
src/java_cup/emit.java:577: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
	  out.println("              return " + pre("do_action_part")+ String.format("%08d",new Integer(0))+"(");
	                                                                                    ^
src/java_cup/emit.java:602: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
	  out.println("              return " + pre("do_action_part")+ String.format("%08d",new Integer(instancecounter))+"(");
	                                                                                    ^
src/java_cup/emit.java:1097: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
		     pre("do_action_part")+ String.format("%08d",new Integer(instancecounter)) +"(");
		                                                 ^
src/java_cup/emit.java:1219: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
	  out.println("              return " + pre("do_action_part")+ String.format("%08d",new Integer(0))+"(");
	                                                                                    ^
src/java_cup/emit.java:1244: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
	  out.println("              return " + pre("do_action_part")+ String.format("%08d",new Integer(instancecounter))+"(");
	                                                                                    ^
src/java_cup/production.java:157: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
      _all.put(new Integer(_index),this);
               ^
src/java_cup/production.java:230: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    return (production) _all.get(new Integer(indx));
                                 ^
src/java_cup/lalr_item_set.java:347: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
	  hashcode_cache = new Integer(result);
	                   ^
src/java_cup/non_terminal.java:43: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
      _all_by_index.put(new Integer(_index), this);
                        ^
src/java_cup/non_terminal.java:93: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
      Integer the_indx = new Integer(indx);
                         ^
src/java_cup/terminal.java:48: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
      _all_by_index.put(new Integer(_index), this);
                        ^
src/java_cup/terminal.java:117: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
      Integer the_indx = new Integer(indx);
                         ^
src/java_cup/runtime/virtual_parse_stack.java:93: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
      vstack.push(new Integer(stack_sym.parse_state));
                  ^
src/java_cup/runtime/virtual_parse_stack.java:140: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
      vstack.push(new Integer(state_num));
                  ^
src/java_cup/runtime/lr_parser.java:480: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
		  ret.add(new Integer(row[i]));
		          ^
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
20 warnings
updated manifest
added manifest
adding: java_cup/runtime/(in = 0) (out= 0)(stored 0%)
adding: java_cup/runtime/ComplexSymbolFactory$ComplexSymbol.class(in = 2816) (out= 1107)(deflated 60%)
adding: java_cup/runtime/ComplexSymbolFactory$Location.class(in = 2529) (out= 1183)(deflated 53%)
adding: java_cup/runtime/ComplexSymbolFactory.class(in = 2327) (out= 586)(deflated 74%)
adding: java_cup/runtime/DefaultSymbolFactory.class(in = 1676) (out= 531)(deflated 68%)
adding: java_cup/runtime/Scanner.class(in = 211) (out= 162)(deflated 23%)
adding: java_cup/runtime/ScannerBuffer.class(in = 967) (out= 528)(deflated 45%)
adding: java_cup/runtime/Symbol.class(in = 1399) (out= 697)(deflated 50%)
adding: java_cup/runtime/SymbolFactory.class(in = 672) (out= 205)(deflated 69%)
adding: java_cup/runtime/SyntaxTreeDFS$AbstractVisitor.class(in = 1706) (out= 760)(deflated 55%)
adding: java_cup/runtime/SyntaxTreeDFS$ElementHandler.class(in = 393) (out= 227)(deflated 42%)
adding: java_cup/runtime/SyntaxTreeDFS$Visitor.class(in = 283) (out= 200)(deflated 29%)
adding: java_cup/runtime/SyntaxTreeDFS.class(in = 1002) (out= 559)(deflated 44%)
adding: java_cup/runtime/SyntaxTreeTransform$1.class(in = 239) (out= 167)(deflated 30%)
adding: java_cup/runtime/SyntaxTreeTransform$ChainRemover.class(in = 1760) (out= 850)(deflated 51%)
adding: java_cup/runtime/SyntaxTreeTransform.class(in = 826) (out= 412)(deflated 50%)
adding: java_cup/runtime/SyntaxTreeXPath.class(in = 3086) (out= 1616)(deflated 47%)
adding: java_cup/runtime/XMLElement$Error.class(in = 1829) (out= 877)(deflated 52%)
adding: java_cup/runtime/XMLElement$NonTerminal.class(in = 3377) (out= 1677)(deflated 50%)
adding: java_cup/runtime/XMLElement$Terminal.class(in = 2612) (out= 1209)(deflated 53%)
adding: java_cup/runtime/XMLElement.class(in = 3509) (out= 1736)(deflated 50%)
adding: java_cup/runtime/lr_parser.class(in = 13959) (out= 7235)(deflated 48%)
adding: java_cup/runtime/virtual_parse_stack.class(in = 1573) (out= 894)(deflated 43%)
>>> Source compiled.
>>> Test phase [not enabled]: dev-java/javacup-11b_p20160615

>>> Install dev-java/javacup-11b_p20160615 into /var/tmp/portage/dev-java/javacup-11b_p20160615/image
debug: java-pkg_dojar: entering function, parameters: javacup.jar
debug: java-pkg_init_paths_: entering function, parameters: 
debug: JAVA_PKG_SHAREPATH: /usr/share/javacup
debug: JAVA_PKG_ENV: /var/tmp/portage/dev-java/javacup-11b_p20160615/image/usr/share/javacup/package.env
debug: JAVA_PKG_JARDEST: /usr/share/javacup/lib
debug: JAVA_PKG_LIBDEST: /usr/lib64/javacup
debug: JAVA_PKG_WARDEST: /usr/share/javacup/webapps
 [33;01m*[0m Unable to perform class version checks as
 [33;01m*[0m class-version-verify.py is unavailable
 [33;01m*[0m Please install dev-java/javatoolkit.
debug: java-pkg_append_: entering function, parameters: JAVA_PKG_CLASSPATH /usr/share/javacup/lib/javacup.jar
debug: installed javacup.jar to /var/tmp/portage/dev-java/javacup-11b_p20160615/image/usr/share/javacup/lib
 [32m*[0m Verbose logging for "java-pkg_dojar" function
 [32m*[0m Jar file(s) destination: /usr/share/javacup/lib
 [32m*[0m Jar file(s) created: javacup.jar
 [32m*[0m Complete command:
 [32m*[0m java-pkg_dojar javacup.jar
debug: java-pkg_do_write_: entering function, parameters: 
debug: java-pkg_init_paths_: entering function, parameters: 
debug: JAVA_PKG_SHAREPATH: /usr/share/javacup
debug: JAVA_PKG_ENV: /var/tmp/portage/dev-java/javacup-11b_p20160615/image/usr/share/javacup/package.env
debug: JAVA_PKG_JARDEST: /usr/share/javacup/lib
debug: JAVA_PKG_LIBDEST: /usr/lib64/javacup
debug: JAVA_PKG_WARDEST: /usr/share/javacup/webapps
debug: java-pkg_dolauncher: entering function, parameters: javacup --main java_cup/Main
debug: java-pkg_init_paths_: entering function, parameters: 
debug: JAVA_PKG_SHAREPATH: /usr/share/javacup
debug: JAVA_PKG_ENV: /var/tmp/portage/dev-java/javacup-11b_p20160615/image/usr/share/javacup/package.env
debug: JAVA_PKG_JARDEST: /usr/share/javacup/lib
debug: JAVA_PKG_LIBDEST: /usr/lib64/javacup
debug: JAVA_PKG_WARDEST: /usr/share/javacup/webapps
debug: java-pkg_dojar: entering function, parameters: javacup.jar javacup-runtime.jar
debug: java-pkg_init_paths_: entering function, parameters: 
debug: JAVA_PKG_SHAREPATH: /usr/share/javacup
debug: JAVA_PKG_ENV: /var/tmp/portage/dev-java/javacup-11b_p20160615/image/usr/share/javacup/package.env
debug: JAVA_PKG_JARDEST: /usr/share/javacup/lib
debug: JAVA_PKG_LIBDEST: /usr/lib64/javacup
debug: JAVA_PKG_WARDEST: /usr/share/javacup/webapps
 [33;01m*[0m Overwriting /var/tmp/portage/dev-java/javacup-11b_p20160615/image/usr/share/javacup/lib/javacup.jar
 [33;01m*[0m Unable to perform class version checks as
 [33;01m*[0m class-version-verify.py is unavailable
 [33;01m*[0m Please install dev-java/javatoolkit.
debug: java-pkg_append_: entering function, parameters: JAVA_PKG_CLASSPATH /usr/share/javacup/lib/javacup.jar
debug: installed javacup.jar to /var/tmp/portage/dev-java/javacup-11b_p20160615/image/usr/share/javacup/lib
 [33;01m*[0m Unable to perform class version checks as
 [33;01m*[0m class-version-verify.py is unavailable
 [33;01m*[0m Please install dev-java/javatoolkit.
debug: java-pkg_append_: entering function, parameters: JAVA_PKG_CLASSPATH /usr/share/javacup/lib/javacup-runtime.jar
debug: installed javacup-runtime.jar to /var/tmp/portage/dev-java/javacup-11b_p20160615/image/usr/share/javacup/lib
 [32m*[0m Verbose logging for "java-pkg_dojar" function
 [32m*[0m Jar file(s) destination: /usr/share/javacup/lib
 [32m*[0m Jar file(s) created: javacup.jar javacup-runtime.jar
 [32m*[0m Complete command:
 [32m*[0m java-pkg_dojar javacup.jar javacup-runtime.jar
debug: java-pkg_do_write_: entering function, parameters: 
debug: java-pkg_init_paths_: entering function, parameters: 
debug: JAVA_PKG_SHAREPATH: /usr/share/javacup
debug: JAVA_PKG_ENV: /var/tmp/portage/dev-java/javacup-11b_p20160615/image/usr/share/javacup/package.env
debug: JAVA_PKG_JARDEST: /usr/share/javacup/lib
debug: JAVA_PKG_LIBDEST: /usr/lib64/javacup
debug: JAVA_PKG_WARDEST: /usr/share/javacup/webapps
>>> Completed installing dev-java/javacup-11b_p20160615 into /var/tmp/portage/dev-java/javacup-11b_p20160615/image

 [32m*[0m Final size of build directory: 1461 KiB (1.4 MiB)
 [32m*[0m Final size of installed tree:   248 KiB

 [32;01m*[39;49;00m checking 7 files for package collisions
>>> Merging dev-java/javacup-11b_p20160615 to /
 [31;01m*[0m Install dev-java/java-dep-check for dependency checking
--- /usr/
--- /usr/share/
>>> /usr/share/javacup/
>>> /usr/share/javacup/package.env
>>> /usr/share/javacup/lib/
>>> /usr/share/javacup/lib/javacup-runtime.jar
>>> /usr/share/javacup/lib/javacup.jar
--- /usr/share/ant/
--- /usr/share/ant/tasks/
>>> /usr/share/ant/tasks/javacup
--- /usr/share/doc/
>>> /usr/share/doc/javacup-11b_p20160615/
>>> /usr/share/doc/javacup-11b_p20160615/html/
>>> /usr/share/doc/javacup-11b_p20160615/html/manual.html
>>> /usr/share/doc/javacup-11b_p20160615/changelog.txt.bz2
--- /usr/bin/
>>> /usr/bin/javacup
>>> dev-java/javacup-11b_p20160615 merged.
```

vaukai commented:
Back to draft. Will try something else.

vaukai commented:
Re-activated for https://bugs.gentoo.org/776655#c5

fordfrog commented:
i'm still getting the issue:
```
Reading "flex/Lexer.jflex"
Exception in thread "main" java.lang.NoSuchMethodError: java_cup.runtime.SymbolFactory.newSymbol(Ljava/lang/String;ILjava_cup/runtime/Symbol;Ljava/lang/Object;)Ljava_cup/runtime/Symbol;
	at jflex.LexParse$CUP$LexParse$actions.CUP$LexParse$do_action_part00000000(LexParse.java:1415)
	at jflex.LexParse$CUP$LexParse$actions.CUP$LexParse$do_action(LexParse.java:3399)
	at jflex.LexParse.do_action(LexParse.java:940)
	at java_cup.runtime.lr_parser.parse(lr_parser.java:584)
	at jflex.Main.generate(Main.java:67)
	at jflex.Main.generate(Main.java:394)
	at jflex.Main.main(Main.java:411)
```
grepping the java file, such method really does not exist:
```
# grep -E "public.*newSymbol" src/java_cup/runtime/SymbolFactory.java
    public Symbol newSymbol(String name, int id, Symbol left, Symbol right, Object value);
    public Symbol newSymbol(String name, int id, Symbol left, Symbol right);
    public Symbol newSymbol(String name, int id, Symbol left, Object value);
    public Symbol newSymbol(String name, int id, Object value);
    public Symbol newSymbol(String name, int id);
```
we need a method with `String, Symbol, Object` parameters but all of those contain `int` as the second parameter. so it seems `jflex` is not compatible with this `javacup` version.

vaukai commented:
> i'm still getting the issue:
> 
> ```
> Reading "flex/Lexer.jflex"
> Exception in thread "main" java.lang.NoSuchMethodError: java_cup.runtime.SymbolFactory.newSymbol(Ljava/lang/String;ILjava_cup/runtime/Symbol;Ljava/lang/Object;)Ljava_cup/runtime/Symbol;
> 	at jflex.LexParse$CUP$LexParse$actions.CUP$LexParse$do_action_part00000000(LexParse.java:1415)
> 	at jflex.LexParse$CUP$LexParse$actions.CUP$LexParse$do_action(LexParse.java:3399)
> 	at jflex.LexParse.do_action(LexParse.java:940)
> 	at java_cup.runtime.lr_parser.parse(lr_parser.java:584)
> 	at jflex.Main.generate(Main.java:67)
> 	at jflex.Main.generate(Main.java:394)
> 	at jflex.Main.main(Main.java:411)
> ```
> 
> grepping the java file, such method really does not exist:
> 
> ```
> # grep -E "public.*newSymbol" src/java_cup/runtime/SymbolFactory.java
>     public Symbol newSymbol(String name, int id, Symbol left, Symbol right, Object value);
>     public Symbol newSymbol(String name, int id, Symbol left, Symbol right);
>     public Symbol newSymbol(String name, int id, Symbol left, Object value);
>     public Symbol newSymbol(String name, int id, Object value);
>     public Symbol newSymbol(String name, int id);
> ```
> 
> we need a method with `String, Symbol, Object` parameters but all of those contain `int` as the second parameter. so it seems `jflex` is not compatible with this `javacup` version.

Thanks for testing.  I try to see what I am getting here.

vaukai commented:
> i'm still getting the issue:
> 
> ```
> Reading "flex/Lexer.jflex"
> Exception in thread "main" java.lang.NoSuchMethodError: java_cup.runtime.SymbolFactory.newSymbol(Ljava/lang/String;ILjava_cup/runtime/Symbol;Ljava/lang/Object;)Ljava_cup/runtime/Symbol;
> 	at jflex.LexParse$CUP$LexParse$actions.CUP$LexParse$do_action_part00000000(LexParse.java:1415)
> 	at jflex.LexParse$CUP$LexParse$actions.CUP$LexParse$do_action(LexParse.java:3399)
> 	at jflex.LexParse.do_action(LexParse.java:940)
> 	at java_cup.runtime.lr_parser.parse(lr_parser.java:584)
> 	at jflex.Main.generate(Main.java:67)
> 	at jflex.Main.generate(Main.java:394)
> 	at jflex.Main.main(Main.java:411)
> ```

Does that happen after `Generate lexer` or after `Regenerate lexer with intermediate jar file`
```
* Generate lexer
Reading "flex/Lexer.jflex"

Warning in file "flex/Lexer.jflex" (line 102): 
".|\n" does not match all characters, because "." excludes all Unicode newline chars - use "[^]" instead
  .|\n            { sb.append(yytext()); }
^

Warning in file "flex/Lexer.jflex" (line 106): 
".|\n" does not match all characters, because "." excludes all Unicode newline chars - use "[^]" instead
.|\n          { emit_warning("Unrecognized character '" +yytext()+"' -- ignored"); }
^
Constructing NFA : 241 states in NFA
Converting NFA to DFA : 
...............................................................................................................................................
147 states before minimization, 132 states in minimized DFA
Writing code to "java/javacup/Lexer.java"

```
```
* Regenerate lexer with intermediate jar file
Reading "flex/Lexer.jflex"

Warning in file "flex/Lexer.jflex" (line 102): 
".|\n" does not match all characters, because "." excludes all Unicode newline chars - use "[^]" instead
  .|\n            { sb.append(yytext()); }
^

Warning in file "flex/Lexer.jflex" (line 106): 
".|\n" does not match all characters, because "." excludes all Unicode newline chars - use "[^]" instead
.|\n          { emit_warning("Unrecognized character '" +yytext()+"' -- ignored"); }
^
Constructing NFA : 241 states in NFA
Converting NFA to DFA : 
...............................................................................................................................................
147 states before minimization, 132 states in minimized DFA
Writing code to "java/javacup/Lexer.java"
```

fordfrog commented:
for `jflex` either the `ant-task` use flag has to be enabled when tests are enabled or the ant task test should not be performed when `ant-task` use flag is disabled:
```
>>> Test phase: dev-java/jflex-1.6.1-r3
javac -source 1.8 -target 1.8 -classpath src/test/java:jflex.jar:lib/java-cup-11a.jar:/usr/share/junit-4/lib/junit.jar src/test/java/jflex/AntTaskTests.java src/test/java/jflex/CharClassesTest.java src/test/java/jflex/EmitterTest.java src/test/java/jflex/IntCharSetTest.java src/test/java/jflex/PackEmitterTest.java src/test/java/jflex/RegExpTests.java src/test/java/jflex/SkeletonTest.java src/test/java/jflex/UnicodePropertiesTest.java
warning: [options] bootstrap class path not set in conjunction with -source 8
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
src/test/java/jflex/AntTaskTests.java:15: error: package jflex.anttask does not exist
import jflex.anttask.JFlexTask;
                    ^
src/test/java/jflex/AntTaskTests.java:28: error: cannot find symbol
  private JFlexTask task;
          ^
  symbol:   class JFlexTask
  location: class AntTaskTests
src/test/java/jflex/AntTaskTests.java:47: error: cannot find symbol
    task = new JFlexTask();
               ^
  symbol:   class JFlexTask
  location: class AntTaskTests
3 errors
4 warnings
```

vaukai commented:
> for `jflex` either the `ant-task` use flag has to be enabled when tests are enabled or the ant task test should not be performed when `ant-task` use flag is disabled:

Ooops, how could I forget about tests?

Done.
